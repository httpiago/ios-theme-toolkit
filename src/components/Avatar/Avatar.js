import React from 'react'
import PropTypes from 'prop-types'
import { mapClassList } from '../../Utils'

/**
 * Component responsible for creating the Avatar.
 *
 * @since 0.1.0
 * @see https://httpiago.github.io/ios-theme-toolkit/#/components/Avatar
 */
export default function Avatar({ src, size, round, className: aditionalClasses, ...rest }) {

  const prefix = 'avatar'
  const classes = mapClassList({
    [`${prefix}--${size}`]: size,
    [`${prefix}--round`]: round
  })

  return (
    <img
      src={src}
      className={`${prefix} ${classes} ${aditionalClasses}`}
      style={(typeof size === 'number') ? { height: size, width: size } : {}}
      {...rest}
    />
  )
}

Avatar.defaultProps = {
  size: 'default',
  round: false,
  className: ''
}

Avatar.propTypes = {
  /** Image url. */
  src: PropTypes.string.isRequired,
  /** Size of avatar. Can be "default", "small", "big" or you can set a custom size by providing a number. */
  size: PropTypes.oneOfType([ PropTypes.number, PropTypes.oneOf([ 'default', 'small', 'big' ]) ]),
  /** Rounded element */
  round: PropTypes.bool,
  /** Aditional classes. */
  className: PropTypes.string,
}
