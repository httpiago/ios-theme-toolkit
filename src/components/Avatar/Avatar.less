@import '../../customizations.less';

img.avatar {
  background-color: rgba(0,0,0,0.12);
  border: none;
  border-radius: @base-border-radius;
  position: relative;
  display: inline-block;
  vertical-align: middle;
  height: 32px;
  width: 32px;
  outline: 0;
}

img.avatar--round {
  border-radius: 50%;
}

img.avatar--small {
  height: 24px;
  width: 24px;
}

img.avatar--big {
  height: 48px;
  width: 48px;
}