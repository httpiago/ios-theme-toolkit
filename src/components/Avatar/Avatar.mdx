---
name: Avatar
route: /components/Avatar
menu: Components
---

import { Playground, PropsTable } from 'docz'
import { Avatar } from '../../index.js'
import '../../../dist/styles.css'

# Avatar

To use this component, simply import it into any file and pass the [modifiers](#/#props) in the properties.

```jsx
import { Avatar } from 'ios-theme-toolkit'

export default () => (
  <Avatar src="" />
)
```

## Examples

<Playground>  
  <span className="desc">Desfault shape and size (32px):</span>
  <Avatar src="https://uinames.com/api/photos/male/6.jpg" />

  <span className="desc">Round edges:</span>
  <Avatar src="https://uinames.com/api/photos/male/6.jpg" round />
  
  <span className="desc">Small (24px):</span>
  <Avatar src="https://uinames.com/api/photos/male/15.jpg" round size="small" />

  <span className="desc">Big (48px):</span>
  <Avatar src="https://uinames.com/api/photos/male/20.jpg" round size="big" />

  <span className="desc">Custom size (~px):</span>
  <Avatar src="https://uinames.com/api/photos/female/6.jpg" round size={150} />
  
  <style>{".avatar { margin: 0 6px}"}</style>
  <style>{".react-live-preview .desc { display: block; font-weight: bold; margin-top: 20px; }"}</style>
</Playground>

## Props

<PropsTable of={ Avatar } />

All the properties which are not listed above will be transferred to the \<img\> tag, so, It accepts all props which native img support.