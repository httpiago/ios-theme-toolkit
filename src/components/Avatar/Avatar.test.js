/**
 * TEST FILE
 * Please try writing the tests for all component behaviors,
 * manually call the functions within the class, change the properties...
 *
 * @see https://jestjs.io/docs/en/expect
 * @see https://github.com/airbnb/enzyme/tree/master/docs/api
 */
import React from 'react'
import { shallow } from 'enzyme'
import Avatar from './Avatar'

const COMPONENT = shallow(
  <Avatar src="https://randomuser.me/api/portraits/women/94.jpg" className="aditionalClass" />
)

describe('Test Avatar component', () => {
  it('should render correctly based on last snapshot', () => {
    expect(COMPONENT).toMatchSnapshot()
  })

  it('should update correctly', () => {
    COMPONENT.setProps({ round: true })

    expect(COMPONENT.hasClass('avatar--round')).toBeTruthy()
  })

  it('should can be a custom size', () => {
    COMPONENT.setProps({ size: 250 })

    expect(COMPONENT.html()).toContain('height:250px')
    expect(COMPONENT.html()).toContain('width:250px')
  })

  it('should has aditional classes', () => {
    expect(COMPONENT.hasClass('aditionalClass')).toBeTruthy()
  })

  it('should pass aditional props to html element', () => {
    COMPONENT.setProps({ 'data-custom-attr': 'yes' })

    expect(COMPONENT.html()).toContain('data-custom-attr="yes"')
  })

  it('should return a img tag', () => {
    expect(COMPONENT.type()).toEqual('img')
  })

})
