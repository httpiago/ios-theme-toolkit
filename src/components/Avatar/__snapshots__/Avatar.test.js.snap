// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Avatar component should render correctly based on last snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Avatar
    className="aditionalClass"
    round={false}
    size="default"
    src="https://randomuser.me/api/portraits/women/94.jpg"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "avatar  aditionalClass",
      "src": "https://randomuser.me/api/portraits/women/94.jpg",
      "style": Object {},
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "avatar  aditionalClass",
        "src": "https://randomuser.me/api/portraits/women/94.jpg",
        "style": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
